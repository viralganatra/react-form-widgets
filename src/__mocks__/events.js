function noop() {}

export const EventTypes = ['onChange', 'onKeyPress', 'onKeyUp', 'onKeyDown', 'onBlur', 'onFocus'];

export const DefaultEventProps = {
    onChange: noop,
    onKeyPress: noop,
    onKeyUp: noop,
    onKeyDown: noop,
    onBlur: noop,
    onFocus: noop,
};
