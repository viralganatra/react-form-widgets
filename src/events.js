function noop() {}

export const EventTypes = ['onChange', 'onKeyPress', 'onKeyUp', 'onKeyDown', 'onBlur', 'onFocus'];

export const DefaultEventProps = EventTypes.reduce((memo, event) => {
    return {
        ...memo,
        [event]: noop,
    };
}, {});
