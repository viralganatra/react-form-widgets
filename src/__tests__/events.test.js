import { EventTypes, DefaultEventProps } from '../events';

describe('events', () => {
    it('should return a array of event types', () => {
        expect(EventTypes).toMatchSnapshot();
    });

    it('should return an object with each event type as a key and a noop function as the value', () => {
        expect(DefaultEventProps).toMatchSnapshot();
    });
});
