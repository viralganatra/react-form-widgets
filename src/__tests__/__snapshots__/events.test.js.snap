exports[`events should return a array of event types 1`] = `
Array [
  "onChange",
  "onKeyPress",
  "onKeyUp",
  "onKeyDown",
  "onBlur",
  "onFocus",
]
`;

exports[`events should return an object with each event type as a key and a noop function as the value 1`] = `
Object {
  "onBlur": [Function],
  "onChange": [Function],
  "onFocus": [Function],
  "onKeyDown": [Function],
  "onKeyPress": [Function],
  "onKeyUp": [Function],
}
`;
